docs/design-documents/20240430-schema-support.md

```diff
@@ -0,0 +1,161 @@
+# Schema support
+
+## The problem
+
+A Conduit user would like to start using a destination connector **without**
+manually setting up
+the [target collections](https://conduit.io/docs/introduction/vocabulary).
+
+To do so, the following is required:
+
+* collection name
+* collection metadata
+* schema
+
+This design document focuses on the schema support in Conduit in the context of
+the above.
+
+## Requirements
+
+1. Records **should not** carry the full schema. 
+
+   Reason: If a record would carry the whole schema, that would increase the
+   record size a lot.
```

Comment:
I think "a lot" is relative to the size of the schema and records. 👍

Reply:
Thanks, my thinking is that there can be very payload heavy records with simple schema and then there can be big schema of ints.. so the runes to describe will probably take more space than the numbers.